plugins/Eq/EqEffect.cpp

```diff
@@ -287,70 +287,61 @@ bool EqEffect::processAudioBuffer( sampleFrame *buf, const fpp_t frames )
 
 
 
-float EqEffect::peakBand( float minF, float maxF, EqAnalyser *fft, int sr )
+float EqEffect::linearPeakBand(float minF, float maxF, EqAnalyser* fft, int sr)
 {
 	auto const fftEnergy = fft->getEnergy();
 	if (fftEnergy == 0.) { return 0.; }
 
-	float peak = -60;
-	float *b = fft->m_bands;
-	float h = 0;
-	for( int x = 0; x < MAX_BANDS; x++, b++ )
+
+	float peakLinear = 0.;
+
+	for (int i = 0; i < MAX_BANDS; ++i)
 	{
-		if( bandToFreq( x ,sr) >= minF && bandToFreq( x,sr ) <= maxF )
+		if (bandToFreq(i, sr) >= minF && bandToFreq(i, sr) <= maxF)
 		{
-			h = 20. * log10(*b / fftEnergy);
-			peak = h > peak ? h : peak;
+			peakLinear = std::max(peakLinear, fft->m_bands[i] / fftEnergy);
 		}
 	}
 
-	return ( peak + 60 ) / 100;
+	return peakLinear;
 }
 
 
 
 
 void EqEffect::setBandPeaks( EqAnalyser *fft, int samplerate )
 {
+	auto computePeakBand = [this, fft, samplerate](FloatModel const & freqModel, FloatModel const & bwModel)
+	{
+		float const freq = freqModel.value();
+		float const bw = bwModel.value();
+
+		return linearPeakBand(freq * (1 - bw * 0.5), freq * (1 + bw * 0.5), fft, samplerate);
+	};
+
 	m_eqControls.m_lowShelfPeakR = m_eqControls.m_lowShelfPeakL =
-			peakBand( m_eqControls.m_lowShelfFreqModel.value()
-					  * ( 1 - m_eqControls.m_lowShelfResModel.value() * 0.5 ),
+			linearPeakBand(m_eqControls.m_lowShelfFreqModel.value()
```

Comment:
This is tabbed twice, we only need once. Similar with the rest of the argument lines below it.

Reply:
Fixed with commit bb212eebbd3.